contracts/interfaces/IOriginationControllerSTIRFRY.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+import "./IOriginationControllerBase.sol";
+
+import "../libraries/LoanLibrary.sol";
+
+interface IOriginationControllerSTIRFRY is IOriginationControllerBase {
+
+    // ============= Data Types =============
+
+    struct StirfryData {
+        address vaultedCurrency;
+        uint256 lenderVaultedCurrencyAmount;
+        uint256 borrowerVaultedCurrencyAmount;
+        uint256 payableToVaultedCurrencyRatio;
+    }
+
+    // ============= Loan Origination =============
+
+    function initializeStirfryLoan(
+        LoanLibrary.LoanTerms calldata loanTerms,
+        StirfryData calldata stirfryData,
+        address borrower,
+        address lender,
+        Signature calldata sig,
+        SigProperties calldata sigProperties,
+        LoanLibrary.Predicate[] calldata itemPredicates
```

Comment:
We don't want to use predicates at all here; it seems like that only confuses things? All the balances we need for the swap are already being checked by this contract.